File: examples/shopflow/sdks/java/Services.java
package com.app.sdk;

import java.util.HashMap;
import java.util.Map;

public final class Services {
    private Services() {}

    public static class DummyJsonService {
        private final Client client;

        public DummyJsonService(Client client) {
            this.client = client != null ? client : new Client();
        }

        public DummyJsonService() {
            this(new Client());
        }

        /** GET products/categories */
        public ApiResponse<String> getCategories() {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "products/categories", pathParams, null, null, null);
        }

        /** GET products/category-list */
        public ApiResponse<String> getCategoryList() {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "products/category-list", pathParams, null, null, null);
        }

        /** GET auth/me */
        public ApiResponse<String> getCurrentUser() {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "auth/me", pathParams, null, null, null);
        }

        /** GET products/{id} */
        public ApiResponse<String> getProductById(String id) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("id", id);
            return this.client.send("GET", "products/{id}", pathParams, null, null, null);
        }

        /** GET products */
        public ApiResponse<String> getProducts(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "products", pathParams, queryParams, null, null);
        }

        /** GET products/category/{category} */
        public ApiResponse<String> getProductsByCategory(String category, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("category", category);
            return this.client.send("GET", "products/category/{category}", pathParams, queryParams, null, null);
        }

        /** POST auth/login */
        public ApiResponse<String> login(String payload) {
            Map<String, String> pathParams = null;
            return this.client.send("POST", "auth/login", pathParams, null, null, payload);
        }

        /** GET products/search */
        public ApiResponse<String> searchProducts(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "products/search", pathParams, queryParams, null, null);
        }

    }

}

File: examples/shopflow/sdks/go/services.go
package sdk

import (
    "context"
)

// DummyJsonService provides access to 8 API endpoints.
type DummyJsonService struct {
    Client *Client
}

// NewDummyJsonService initializes a new DummyJsonService
func NewDummyJsonService(client *Client) *DummyJsonService {
    return &DummyJsonService{Client: client}
}

// GetCategories: GET products/categories
func (s *DummyJsonService) GetCategories(ctx context.Context) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "products/categories", pathParams, nil, nil, nil)
}

// GetCategoryList: GET products/category-list
func (s *DummyJsonService) GetCategoryList(ctx context.Context) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "products/category-list", pathParams, nil, nil, nil)
}

// GetCurrentUser: GET auth/me
func (s *DummyJsonService) GetCurrentUser(ctx context.Context) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "auth/me", pathParams, nil, nil, nil)
}

// GetProductById: GET products/{id}
func (s *DummyJsonService) GetProductById(ctx context.Context, id string) (*ApiResponse, error) {
    pathParams := map[string]string{
        "id": id,
    }
    return s.Client.DoRequest(ctx, "GET", "products/{id}", pathParams, nil, nil, nil)
}

// GetProducts: GET products
func (s *DummyJsonService) GetProducts(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "products", pathParams, queryParams, nil, nil)
}

// GetProductsByCategory: GET products/category/{category}
func (s *DummyJsonService) GetProductsByCategory(ctx context.Context, category string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "category": category,
    }
    return s.Client.DoRequest(ctx, "GET", "products/category/{category}", pathParams, queryParams, nil, nil)
}

// Login: POST auth/login
func (s *DummyJsonService) Login(ctx context.Context, payload interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "POST", "auth/login", pathParams, nil, nil, payload)
}

// SearchProducts: GET products/search
func (s *DummyJsonService) SearchProducts(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "products/search", pathParams, queryParams, nil, nil)
}
